examples/url-shortener/src/lib.rs

```diff
@@ -0,0 +1,64 @@
+#[macro_use]
+extern crate rocket;
+
+use rocket::{
+    response::{status::BadRequest, Redirect},
+    routes, Build, Rocket, State,
+};
+use serde::Serialize;
+use shuttle_service::Factory;
+use sqlx::{Executor, FromRow, PgPool};
+
+#[macro_use]
+extern crate shuttle_service;
+
+struct AppState {
+    pool: PgPool,
+}
+
+#[derive(Serialize, FromRow)]
+struct Url {
+    pub id: String,
+    pub url: String,
+}
+
+#[get("/<id>")]
+async fn redirect(id: String, state: &State<AppState>) -> Result<Redirect, BadRequest<String>> {
+    let url: Url = sqlx::query_as("SELECT * FROM urls WHERE id = $1")
+        .bind(id)
+        .fetch_one(&state.pool)
+        .await
+        .map_err(|e| BadRequest(Some(e.to_string())))?;
```

Comment:
On getting a shortened URL which does not exist, you get the plain sqlx response ("no rows returned...") wrapped up in a 400.

Would it not be better to dispatch according:
- `Error:: RowNotFound`: 404 -> "the requested shortened URL does not exist"
- `otherwise`: 500 -> "something went wrong, sorry 🤷" and log out the error